runtime/common/src/main/kotlin/kotlinx/serialization/json/Json.kt

```diff
@@ -1,27 +1,54 @@
 /*
- * Copyright 2018 JetBrains s.r.o.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- * http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
+ * Copyright 2017-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
  */
 package kotlinx.serialization.json
 
 import kotlinx.serialization.*
-import kotlinx.serialization.context.SerialContext
-import kotlinx.serialization.context.SerialModule
+import kotlinx.serialization.context.*
 import kotlinx.serialization.json.internal.*
 import kotlin.jvm.*
 
-class Json(
+/**
+ * The main entry point to work with JSON serialization.
+ * Is is typically used by constructing an application-specific instance, registering
+ * custom serializers via [Json.install] and then using it either as regular [SerialFormat] or [StringFormat]
+ * or for converting objects to [JsonElement] back and forth.
+ *
+ * This is the only serial format which has first-class [JsonElement] support.
+ * Any serializable class can be serialized to or from [JsonElement] with [Json.fromJson] and [Json.toJson] respectively or
+ * serialize properties of [JsonElement] type.
+ *
+ * Json configuration parameters:
+ * [unquoted] specifies keys and values should be quoted, used mostly for testing.
+ * [indented] specifies whether resulting JSON should be pretty-printed.
+ * [indent] specifies which indent string to use with [indented] mode.
+ * [strictMode] enables strict mode, which prohibits unknown keys and infinite values in floating point numbers.
+ *
+ * Example of usage:
+ * ```
+ * @Serializable
+ * class DataHolder(val id: Int, val data: String, val extensions: JsonElement)
```

Comment:
All samples here use `@ImplicitReflectionSerializer` functionality, which is discouraged in general. For this concrete sample it is probably ok (in order not to repeat `DataHolder.serializer()` everywhere), but better add a note about it.